google/gax/api_callable.py

```diff
@@ -41,59 +41,63 @@
 _MILLIS_PER_SECOND = 1000
 
 
-def _add_timeout_arg(a_func, timeout):
-    """Updates a_func so that it gets called with the timeout as its final arg.
+class _RetryableWrapper(object):
```

Comment:
I could rewrite as decorator-style, but I think the code structure is mostly same and using callable class is much more straightforward in this case.

The point is that the decoration (the behavior) may change upon the invocation because it needs to change either of _add_timeout_arg or _retry, and they share some values like a_func and base settings. This is naturally a class, although we could implement it as internal functions.